var/spack/repos/builtin/packages/workrave/package.py

```diff
@@ -0,0 +1,108 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+from shutil import copyfile
+
+
+class Workrave(AutotoolsPackage):
+    """Workrave is a program that assists in the recovery and prevention of
+       Repetitive Strain Injury (RSI). The program frequently alerts you to
+       take micro-pauses, rest breaks and restricts you to your daily limit.
+       The program runs on GNU/Linux and Microsoft Windows.
+    """
+
+    homepage = "http://www.workrave.org/"
+    url      = "https://github.com/rcaelers/workrave/archive/v1_10_20.tar.gz"
+
+    version('1_10_20', '095567c10311bd2c1a52f98035cc8590')
+    version('1_10_19', 'a87ed53d5b321133e8b6b98fd715507b')
+    version('1_10_18', 'd36c2aba0485116b831d5b36a862f343')
+    version('1_10_17', 'ba829bb2c0ec999b3194b4d845549c39')
+    version('1_10_16', 'b9bf77bfe0c909fff56759e7dda40f9d')
+    version('1_10_15', '4a70c2e325503249d1a40fcc236d6802')
+    version('1_10_14', '67108d993719d9438a1b69f0cb8fc9b8')
+    version('1_10_13', 'd5e7110dfb0b0a31c909405913ac2a75')
+    version('1_10_12', '0bfbaa1dc35901ffa8f1a3676421a992')
+    version('1_10_10', 'cf827672c8a1ece074f8ddfcf73d0fe2')
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('libtool',  type='build')
+    depends_on('m4',       type='build')
+
+    depends_on('libx11')
+    depends_on('py-cheetah')
+    depends_on('glib')
+    depends_on('glibmm')
+    depends_on('gtkplus')
+    depends_on('gtkmm@2.17.1')
+    depends_on('libsigcpp')
+
+    # adds #include <time.h> to a workrave test
+    patch('add_time_header.patch')
+
+    # removes call to missing glimm api function
+    patch('signal_size_changed.patch', when='@1.10.0')
```

Comment:
@mmpozulp I'm having trouble finding other instances of this. As far as you know, has this only happened on your system? If so, do you have a notion of why? I'm concerned this means that something is missing or some component needs to be enabled that currently isn't

Reply:
@scheibelp I'm comfortable deleting this patch. It's only necessary when building the workrave tarball that comes from sourceforge (which was what I encountered first when googling for workrave, but am no longer using). Since then I discovered a workrave repo on github. I am using the github repo as the package url, **not** the sourceforge url, partly because of issues like this. The sourceforge tarball src calls nonexistent functions, like ```Gtk::StatusIcon::property_embedded()```. I looked for it in several versions of gtkmm and couldn't find it.

Reply:
Got it - I will wait for that update. Are the other custom patches similarly resolved by changing the source?

Reply:
No. I just tried commenting-out the other 3 patch() calls one-at-a-time and each resulted in an error:

1.
```python 
#patch('add_time_header.patch')
```
```console
  >> 829    DayTimePred.cc:136:9: error: use of undeclared identifier 'localtime'
     830      ret = localtime(&last_time);
```

2.
```python 
#patch('dont_get_widget.patch')
```
```console
  >> 2631    ../../plugin/statistics/gtkmm/src/StatisticsDialog.cc:731:12: error: no member named 'get_widget_for_response' in 'Gtk::MessageDialog'
     2632        mb_ask.get_widget_for_response( Gtk::RESPONSE_NO )->grab_default();
```
3.
```python
#patch('no_gettext.patch')
```
```console
  >> 393    config.status: error: po/Makefile.in.in was not created by intltoolize.
```
Thus, I will leave in the remaining 3 workrave patches.